### PR TITLE
Add RegistryProfile to admin API

### PR DESCRIPTION
**Pull request by petrkotas**
Add registryProfile to the admin api

Fix: #685 

Signed-off-by: Petr Kotas <pkotas@redhat.com>

### CODE REVIEW

File: pkg/api/admin/openshiftcluster.go [hunk 1]
```diff
@@ -163,3 +164,9 @@ const (
 	InstallPhaseBootstrap InstallPhase = iota
 	InstallPhaseRemoveBootstrap
 )
+
+// RegistryProfile represents a registry profile
+type RegistryProfile struct {
+	Name    string `json:"name,omitempty"`
+	UseName string `json:"userName,omitempty"`
```
Comment (mjudeikis) [inline pkg/api/admin/openshiftcluster.go:171]: `UserName`

File: pkg/api/admin/openshiftcluster_convert.go [hunk 2]
```diff
@@ -92,6 +92,14 @@ func (c *openShiftClusterConverter) ToExternal(oc *api.OpenShiftCluster) interfa
 		}
 	}
 
+	if oc.Properties.RegistryProfiles != nil {
+		out.Properties.RegistryProfiles = make([]RegistryProfile, len(oc.Properties.RegistryProfiles))
+		for i, v := range oc.Properties.RegistryProfiles {
+			out.Properties.RegistryProfiles[i].Name = v.Name
+			out.Properties.RegistryProfiles[i].UseName = v.Username
```
Comment (mjudeikis) [inline pkg/api/admin/openshiftcluster_convert.go:99]: ditto

File: pkg/api/admin/openshiftcluster_convert.go [hunk 3]
```diff
@@ -175,4 +183,13 @@ func (c *openShiftClusterConverter) ToInternal(_oc interface{}, out *api.OpenShi
 			Phase: api.InstallPhase(oc.Properties.Install.Phase),
 		}
 	}
+
```
Comment (mjudeikis) [inline pkg/api/admin/openshiftcluster_convert.go:186]: I dont think we need this in the ToInternal. We would not accept changes to this from admin

File: pkg/api/admin/openshiftcluster.go [hunk 4]
```diff
@@ -163,3 +164,9 @@ const (
 	InstallPhaseBootstrap InstallPhase = iota
 	InstallPhaseRemoveBootstrap
 )
+
+// RegistryProfile represents a registry profile
+type RegistryProfile struct {
+	Name     string `json:"name,omitempty"`
+	UserName string `json:"userName,omitempty"`
```
Comment (jim-minter) [inline pkg/api/admin/openshiftcluster.go:171]: must match the internal API exactly:
```
Username string `json:"username,omitempty"`
```

File: pkg/api/admin/openshiftcluster_convert.go (hunk 3 above)
Comment (jim-minter) [reply pkg/api/admin/openshiftcluster_convert.go]: I disagree.  We aim for 100% coverage on the converters in both directions; the validation will prevent any changes being made.

**Comment by mjudeikis** [comment]
Very close. Still need few bits:
1. As per Jims comments - converted both ways.
2. Validation should prevent any changes to be submitted to this object from external API (see production API `v20200430`)
3. Test to validate we can't change those fields.

File: pkg/api/admin/openshiftcluster.go [hunk 5]
```diff
@@ -163,3 +164,9 @@ const (
 	InstallPhaseBootstrap InstallPhase = iota
 	InstallPhaseRemoveBootstrap
 )
+
+// RegistryProfile represents a registry profile
+type RegistryProfile struct {
+	Name     string `json:"name,omitempty"`
+	UserName string `json:"username,omitempty"`
```
Comment (mjudeikis) [inline pkg/api/admin/openshiftcluster.go:171]: `Username string       `json:"username,omitempty"``

**Pull request merged**